src/crewai/task.py

```diff
@@ -115,6 +115,33 @@ class Task(BaseModel):
     _thread: Optional[threading.Thread] = PrivateAttr(default=None)
     _execution_time: Optional[float] = PrivateAttr(default=None)
 
+    @model_validator(mode="before")
+    @classmethod
+    def process_config_and_validate(cls, values):
+        config = values.get("config", {})
+
+        # Process config
+        for key, value in config.items():
```

Comment:
Hm, pretty similar to the config from above, right!? Should we centralize it somehow?

Reply:
I'll move both to a utils file.